source/Calamari/Deployment/PackageRetention/Repositories/JsonJournalRepository.cs

```diff
@@ -14,18 +15,19 @@ namespace Calamari.Deployment.PackageRetention.Repositories
     public class JsonJournalRepository
         : IJournalRepository
     {
+
         Dictionary<PackageIdentity, JournalEntry> journalEntries;
         const string SemaphoreName = "Octopus.Calamari.PackageJournal";
 
         readonly ICalamariFileSystem fileSystem;
-        readonly ISemaphoreFactory semaphoreFactory;
         readonly string journalPath;
+        readonly IDisposable semaphore;
 
-        public JsonJournalRepository(ICalamariFileSystem fileSystem, ISemaphoreFactory semaphoreFactory)
+        public JsonJournalRepository(ICalamariFileSystem fileSystem, ISemaphoreFactory semaphoreFactory, string journalPath)
         {
             this.fileSystem = fileSystem;
-            this.semaphoreFactory = semaphoreFactory;
-            this.journalPath = @"C:\Octopus\PackageJournal.json";//journalPath;
+            this.journalPath = journalPath;
+            this.semaphore = semaphoreFactory.Acquire(SemaphoreName, "Another process is using the package retention journal");
```

Comment:
I was going to suggest that we should move the semaphore factory to the JsonJournalRespository, but given that this repository disposes the semaphore object, I think this is the correct implementation 👍